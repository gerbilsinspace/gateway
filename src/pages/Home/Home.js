import React from 'react';
import Async from 'react-promise';
import Header from '../../components/HeaderContainer';
import Footer from '../../components/FooterContainer';
import Menu from '../../components/Menu';
import { url } from '../../config';
import { getData } from '../../helpers';

import './style.css';

const CtaMap = ({ link, title, deck, image }, index) => (
  <li key={index}>
    <a href={link}>
      <h2>{title}</h2>
      <p>{deck}</p>
      <img src={url + image} alt='' />
    </a>
  </li>
);

const QuoteMap = ({quote, author}, index) => (
  <li key={index}>
    <blockquote>
      <p>{quote}</p>
    </blockquote>
    <p>{author}</p>
  </li>
);

const Description = ({colour, text, image}) => (
  <div>
    <p className={colour}>
      {text}
    </p>
    <div
      className={colour}
      style={{
        backgroundImage: 'url(' + url + image + ')',
        height: '500px'
      }}
    />
  </div>
);

const Home = () => (
  <Async
    promise={new Promise(async (resolve) => {
      const data = await getData(url, 'data/homepage.json');

      resolve({
        header: data.pageData.header,
        menu: {
          menuItems: data.menuData.menu,
          iconUrl: data.iconUrl
        },
        cta: data.pageData.cta,
        quotes: data.pageData.quotes,
        deck: data.pageData.deck
      });
    })}

    then={({header, menu, cta, quotes, deck}) => (
      <div>
        <Header
          text={header.title}
          image={url + header.image}
          colour={header.menuColour}
        />
        <Menu
          menuItems={menu.menuItems}
          menuColour={header.menuColour}
          iconUrl={menu.iconUrl}
        />
        <div className="clearfix" />
        <ul className="call-to-actions">
          { cta.map(CtaMap) }
        </ul>
        <ul className="quotes">
          { quotes.map(QuoteMap) }
        </ul>
        <Description
          colour={deck.colour}
          text={deck.text}
          image={deck.image}
        />
        <Footer />
      </div>
    )}
  />
);

export default Home;
