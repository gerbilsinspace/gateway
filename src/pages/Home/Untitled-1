

title
set font
font size
font colour
keep text centred


mobile first
tablet second
desktop third