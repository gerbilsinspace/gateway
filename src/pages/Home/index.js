import React from 'react';
import Header from '../../components/HeaderContainer';
import Menu from '../../components/Menu';
import homepageData from '../../data/homepage.json';
import menuData from '../../data/menu.json';

import './style.css';

const ctaMap = ({ link, title, deck, image }) => (
  <a href={link}>
    <h2>{title}</h2>
    <p>{deck}</p>
    <img src={image} alt='' />
  </a>
);

const quotesMap = ({quote, author}) => (
  <div>
    <blockquote>
      <p>{quote}</p>
    </blockquote>
    <p>{author}</p>
  </div>
);

const Description = ({colour, text, image}) => (
  <div>
    <p className={colour}>
      {text}
    </p>
    <div
      className={colour}
      style={{
        backgroundImage: 'url(' + image + ')',
        height: '500px'
      }}
    />
  </div>
);

const Home = () => {
  const {
    deck,
    title,
    headerImage,
    menuColour,
    cta,
    quotes
  } = homepageData;
  const {colour, text, image} = deck;

  return (
    <div>
      <Header text={title} image={headerImage} />
      <Menu menu={menuData.menu} menuColour={menuColour} />
      { cta.map(ctaMap) }
      { quotes.map(quotesMap) }
      <Description
        colour={colour}
        text={text}
        image={image}
      />
    </div>
  );
};

export default Home;
