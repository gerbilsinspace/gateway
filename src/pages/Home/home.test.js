import React from 'react';
import { shallow } from 'enzyme';

import Home from './index';

let component;

describe('Homepage', () => {
  beforeEach(() => {
    component = shallow(<Home />);
  });

  it('renders without crashing', () => {
    expect(component).toBeDefined();
  });

  it('contains a header', () => {
    expect(component.find('HeaderContainer')).toExist();
  });

  it('contains a menu', () => {
    expect(component.find('Menu')).toExist();
  });

  it('contains a description', () => {
    expect(component.find('Description')).toExist();
  });

  describe('contains a list of call to actions', () => {
    let ctas;

    beforeEach(() => {
      ctas = component.find('ul.call-to-actions');
    });

    it('should have an unorderedList', () => {
      expect(ctas).toExist();
    });

    it('should not have any items when there is no data', () => {
      console.log(ctas.debug());
      expect(ctas.find('li')).not.toExist();
    });
  });

  it('contains a footer', () => {
    expect(component.find('FooterContainer')).toExist();
  });
});
