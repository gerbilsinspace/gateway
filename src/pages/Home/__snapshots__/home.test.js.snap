// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage after data has loaded should match snapshot 1`] = `
<div>
  <HeaderContainer
    image="https://gateway-cms.netlify.com/undefined"
  />
  <Menu
    iconUrl="https://gateway-cms.netlify.com/uploads/gateway_logo_pink.png"
    menuColour="red"
  />
  <ul
    className="call-to-actions"
  />
  <ul
    className="quotes"
  />
  <Description />
  <FooterContainer />
</div>
`;

exports[`Homepage before data has loaded should match snapshot 1`] = `<Loading />`;
